### PR TITLE
Add WMTS temporal settings

### LINKED ISSUE #6614: Cleanup and simplify WMS-T settings and make it easier to select the desired time for some providers (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#42399
Author: @nyalldawson
QGIS version: 3.20

**Cleanup and simplify WMS-T settings and make it easier to select the desired time for some providers**

### PR Description:
(Temporarily includes the commits from the pre-requisite https://github.com/qgis/QGIS/pull/42372 PR)

This PR reworks how WMS-T settings are exposed to users, in order to:
1. Make it clearer exactly what each of the settings controls
2. Make the interplay between the settings clearer
3. Make it obvious which settings apply regardless of whether the layer is controlled by the temporal controller vs the settings which relate only to static wms-t layers
4. And lastly, for servers which expose a non-contiguous set of date time instances (instead of a range of dates) we now present the available options as a combobox of valid choices instead of requiring users to manually enter the valid dates themselves.

Here's how things look with a server exposing a range of available dates:

Peek 2021-03-23 16-03

(this also shows the improved display of reference time, where users can see the valid range of reference time extent in a label)

And here's how things look when the server offers a list of available datetime instances only:

Peek 2021-03-23 16-04

Refs Natural resources Canada Contract: 3000720707

### Commits tagged with [need-docs] or [FEATURE]
"[FEATURE] For wms-t layers which expose a non-contiguous set of\navailable temporal instances, show a combo box in the layer\ntemporal properties with available times instead of requiring\nusers to manually enter date time ranges themselves\n\nRefs Natural resources Canada Contract: 3000720707"

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #6614 and numerous unreported reports
@samweli @nyalldawson Worth checking if anything missing or incorrect. Thanks.

### CODE REVIEW

File: docs/user_manual/working_with_ogc/ogc_client_support.rst [hunk 1]
```diff
@@ -501,6 +501,59 @@ definitions:
...
+.. _figure_wmts_temporal:
+
+.. figure:: img/temporal_properties.png
+   :align: center
+
+   Temporal properties of a WMTS layer
+
+
+Whatever temporal data control is in use, there are some :guilabel:`WMS-T Settings`
+to help display accurate data:
```
Comment (nyalldawson) [inline docs/user_manual/working_with_ogc/ogc_client_support.rst:538]:
```suggestion
to help display the correct temporal data:
```

File: docs/user_manual/working_with_ogc/ogc_client_support.rst [hunk 2]
```diff
@@ -501,6 +501,59 @@ definitions:
...
+
+* :guilabel:`Time slice mode` which can be:
+
+  * :guilabel:`Use whole temporal range`
+  * :guilabel:`Match to start of range`
+  * :guilabel:`match to end of range`
```
Comment (nyalldawson) [inline docs/user_manual/working_with_ogc/ogc_client_support.rst:544]:
```suggestion
  * :guilabel:`Match to end of range`
```

**Comment by nyalldawson** [comment]
LGTM!

**Comment by Samweli** [comment]
@DelazJ @nyalldawson seeing this now! Looks good to me too!

**Pull request merged**